pkg/infra/httpclient/httpclientprovider/host_redirect_validation_middleware.go

```diff
@@ -27,7 +29,7 @@ func RedirectLimitMiddleware(reqValidator validations.PluginRequestValidator) sd
 					return nil, locationErr
 				}
 
-				if validationErr := reqValidator.Validate(location.String(), nil); validationErr != nil {
+				if validationErr := reqValidator.Validate(&datasources.DataSource{URL: location.String()}, setting.SecureSocksDSProxySettings{}, nil); validationErr != nil {
```

Comment:
It seems like this is the only place that is problematic 😞 The only escape hatch we have here is that we should have the following labels available via `opts sdkhttpclient.Options`:

- Labels["datasource_name"]
- Labels["datasource_uid"] 
- Labels["datasource_type"]

They _should_ always be populated, but it would probably be a good idea to actually add types/funcs for these on the SDK struct itself (could be done in a follow up). That means we'd have to read the datasource from the DB in this case though. Maybe that's okay since this is only when status code is 3xx.

Aside: I also think we could get away with not having to pass `setting.SecureSocksDSProxySettings` actually since `*cfg.Setting` could just become a field over [here](https://github.com/grafana/grafana-enterprise/pull/7510/files#diff-a063f9b3aa7eb7f1806ac654408a509e32c132241e53b563f54dcb6ce64f2fffR46) - since that is the only implementation where it's actually used.

Reply:
Regarding the 1st point I wonder if it's ever possible the `res.Location()` (which is what we currently check) to be different from the datasource URL?
Regarding the 2nd point I agree: that would simplify it a lot